### PR TITLE
Implement Integer#digits

### PR DESCRIPTION
**Pull request by havenwood**

### CODE REVIEW

File: src/main/ruby/core/integer.rb [hunk 1]
```diff
@@ -180,6 +180,12 @@ def pred
     self - 1
   end
 
+  def digits base = 10
```
Comment (eregon) [inline src/main/ruby/core/integer.rb]: Could you add `()` around method arguments for clarity?

**Comment by eregon** [comment]
Does this pass all specs?
You can run them with:
```
$ jt test --unguarded spec/ruby/core/integer/digits_spec.rb
```

File: src/main/ruby/core/integer.rb [hunk 2]
```diff
@@ -180,6 +180,12 @@ def pred
...
+    raise Math::DomainError, 'out of domain' if negative?
+
+    to_s(base.to_int).chars.map(&:to_i).reverse
```
Comment (eregon) [inline src/main/ruby/core/integer.rb]:
```ruby
base = Truffle::Type.coerce_to_int(base)
```
is slightly better than just calling `to_int`, as it checks the result is a Fixnum as expected and throws TypeError if coercion doesn't work.

**Comment by havenwood** [comment]
Yes, all the specs pass when run unguarded. I've made your suggested changes and rebased. Thanks!

**Comment by eregon** [comment]
FWIW, the Integer#digits specs are now run in CI thanks to #941.

**Pull request merged**